openeo/extra/job_management/__init__.py

```diff
@@ -364,19 +373,30 @@ def run_loop():
             # TODO: support user-provided `stats`
             stats = collections.defaultdict(int)
 
-            while (
-                sum(job_db.count_by_status(statuses=["not_started", "created", "queued", "running"]).values()) > 0
-                and not self._stop_thread
-            ):
-                self._job_update_loop(job_db=job_db, start_job=start_job)
+            # TODO: multiple workers instead of a single one? Work with thread pool?
+            worker_thread = _JobManagerWorkerThread(work_queue=self._work_queue, result_queue=self._result_queue)
```

Comment:
In my initial sketch I just set this up with a single worker thread, but as we are aiming for bigger use cases, I think we should already go for a pool of workers from the start